In [1]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt

In [2]:
#Create DataFrame
# Data penjualan
data = [100, 150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 50, 100, 100,
        140, 90, 150, 50, 90, 120, 100, 110, 75, 65]

# Membuat DataFrame
df = pd.DataFrame(data, columns=['TransactionAmount'])

# Menampilkan DataFrame
print(df)

    TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65


In [3]:
#Central Tendency
# Menghitung mean
mean_value = df['TransactionAmount'].mean()

# Menghitung median
median_value = df['TransactionAmount'].median()

# Menghitung mode
mode_value = df['TransactionAmount'].mode()


# Menampilkan hasil
print(f"Mean (Rata-Rata): {mean_value}")
print(f"Median: {median_value}")
print(f"Mode: {mode_value.values}")

Mean (Rata-Rata): 102.6
Median: 100.0
Mode: [100]


Interpretasi Hasil:

- Mean (Rata-Rata): 105.0 menunjukkan bahwa nilai rata-rata transaksi adalah 105.

- Median: 100.0 menunjukkan setengah dari transaksi lebih rendah dari 100 dan setengah lebih tinggi dari 100.
      
- Mode: [100] menunjukkan bahwa nilai 100 muncul paling sering dibandingkan nilai lainnya.

In [7]:
# Variability
# Menghitung rentang (range)
data_range = df['TransactionAmount'].max() - df['TransactionAmount'].min()

# Menghitung variansi
variance = df['TransactionAmount'].var()

# Menghitung deviasi standar (standard deviation)
std_dev = df['TransactionAmount'].std()

# Menghitung interquartile range (IQR)
Q1 = df['TransactionAmount'].quantile(0.25)
Q3 = df['TransactionAmount'].quantile(0.75)
IQR = Q3 - Q1

# Menampilkan hasil
print(f"Range (Rentang): {data_range}")
print(f"Variance (Variansi): {variance}")
print(f"Standard Deviation (Deviasi Standar): {std_dev}")
print(f"Interquartile Range (IQR): {IQR}")

Range (Rentang): 100
Variance (Variansi): 1012.7499999999997
Standard Deviation (Deviasi Standar): 31.82373328193912
Interquartile Range (IQR): 35.0


Range               

Memberikan gambaran umum tentang seberapa besar variasi total dalam data. 
Rentang 100 menunjukkan perbedaan yang signifikan antara nilai maksimum dan minimum.

Variance

Mengukur seberapa tersebar nilai-nilai dari rata-rata dalam unit kuadrat.
Variansi yang tinggi menunjukkan data yang sangat tersebar.
    
Standard Deviation

Mengukur deviasi rata-rata dari rata-rata dalam unit yang sama dengan data. 
Deviasi standar 31.82 menunjukkan penyebaran data yang cukup luas di sekitar rata-rata.
IQR (Interquartile Range): Menunjukkan rentang di mana 50% data berada,
dengan IQR 35.0 menunjukkan variasi yang signifikan dalam data di sekitar median.

In [8]:
#H0 = 100
#H1 > 100
#Define Alpha 0.05

In [9]:
len(df)

25

In [11]:
stats.ttest_1samp(a=df, popmean=100)

TtestResult(statistic=array([0.40850016]), pvalue=array([0.68652848]), df=array([24]))

In [ ]:
#statistic= 0.40850016 , pvalue= 0.68652848

In [11]:
#conclude with Alpha and P-value
#as the P-Value is Larger the Alpha (5%), then we Fail to Reject H0

In [13]:
#mencari critical ragion 
#conclude t-statistics with crirical region

stats.t.ppf(1-0.05,26)

1.7056179197592727

In [40]:
#H1 > 100

Critical region result
1.7056179197592727
The Average is still 100 